extensions/indexes/range/src/org/exist/indexing/range/conversion/DateConverter.java

```diff
@@ -42,13 +42,13 @@
 
     protected static final Logger LOG = LogManager.getLogger(DateConverter.class);
 
-    private final static Pattern DATE_REGEX = Pattern.compile("(\\d+)-(\\d+)-(\\d+)");
+    private final static Pattern DATE_REGEX = Pattern.compile("(-?\\d+)-(\\d+)-(\\d+)");
 
     @Override
     public Field toField(String fieldName, String content) {
         try {
             DateValue dv;
-            if (content.indexOf('-') < 0) {
+            if (content.indexOf('-') < 1) {
```

Comment:
You mean like `!(content.substring(1).contains('-'))`? As this should only evaluate false if the string contains '-' in any other position than the first position, accounting for negative "only-year" values.

Seems more complicated to me, but if you insist... 😉